### PR TITLE
chore: add catalog-info.yml

### PR DESCRIPTION
**Pull request by ichintanjoshi**
## Description

- Add catalog-info.yaml

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ichintanjoshi! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  description: >
+          A helper library for mathematical calculations and
+          symbolic mathematics, used by the edx-platform.
+  links:
+    - url: "https://github.com/openedx/openedx-calc"
+      title: "openedx/openedx-calc"
+      icon: "GitHub"
+  annotations:
+    openedx.org/arch-interest-groups: ""
+    openedx.org/release: "main"
```
Comment (feanil) [inline catalog-info.yaml:17]: @ichintanjoshi you need to look at the existing `openedx.yaml` to determine whether or not a repo needs this annotation.  This repo was not getting tagged previously so it should not get tagged now. 
```suggestion
```

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+kind: Component
+metadata:
+  name: 'openedx-calc'
+  description: >
+          A helper library for mathematical calculations and
+          symbolic mathematics, used by the edx-platform.
+  links:
+    - url: "https://github.com/openedx/openedx-calc"
+      title: "openedx/openedx-calc"
+      icon: "GitHub"
```
Comment (feanil) [inline catalog-info.yaml:14]:
```suggestion
  links: []
```

**Review by feanil — CHANGES_REQUESTED** [review]

**Comment by ichintanjoshi** [comment]
Updated as per suggestion @feanil

**Review by feanil — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@ichintanjoshi 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/openedx-calc&entry.867055334=https://github.com/openedx/openedx-calc/pull/104&entry.1484655318=https://github.com/ichintanjoshi&entry.752974735=2024-05-30+07:39&entry.1917517419=2024-08-06+13:28&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**